### PR TITLE
Added all needed files

### PR DESCRIPTION
**Pull request by Scriptkiddi**
This commit adds a completley configurable redis config file
every option is supported.
Also all defaults from the config file are now in defaults/main.yml

### CODE REVIEW

File: defaults/main.yml [hunk 1]
```diff
@@ -0,0 +1,61 @@
...
+redis_repl_backlog_ttl: 3600
+redis_bind: 127.0.0.1
+redis_protected_mode: yes
+redis_port: 6379
+redis_tcp_backlog: 511
+redis_timeout: 0
+redis_tcp_keepalive: 300
+redis_daemonize: no
+redis_supervised: no
+redis_pidfile: /var/run/redis_6379.pid
```
Comment (dasJ) [inline defaults/main.yml]: What is this 6379?

File: README.md [hunk 2]
```diff
@@ -1,27 +1,34 @@
...
-For instance, if the role uses the EC2 module or depends on other Ansible roles, it may be a good idea to mention in this section that the boto package is required.
+An dpkg- or pacman-based Linux distribution.
 
 
 ## Role Variables
 
-A description of the settable variables for this role should go here, including any variables that are in defaults/main.yml, vars/main.yml, and any variables that can/should be set via parameters to the role.
-Any variables that are read from other roles and/or the global scope (ie. hostvars, group vars, etc.) should be mentioned here as well.
-
+This role seriously has a ton of variables. Instead of copying the defaults file here, look it up there. All variables from redis.conf are called exactly like they are called in the file but with redis_ prepended.
```
Comment (dasJ) [inline README.md]: @SlothOfAnarchy What do you think?

File: README.md [hunk 3]
```diff
@@ -1,27 +1,34 @@
...
 
-A brief description of the role goes here.
+This role installs and configures a redis server. It allows to set all configuration variables supported by redis.
 
 
 ## Requirements
 
-Any pre-requisites that may not be covered by Ansible itself or the role should be mentioned here.
-For instance, if the role uses the EC2 module or depends on other Ansible roles, it may be a good idea to mention in this section that the boto package is required.
+An dpkg- or pacman-based Linux distribution.
```
Comment (dasJ) [inline README.md]: I was really happy when I thought you had pacman in mind.... 😞

File: README.md [hunk 4]
```diff
@@ -1,27 +1,34 @@
 # Role Name
```
Comment (SlothOfAnarchy) [inline README.md]: I think we have enough `Role Name` roles in this group already, do we really need this repo?

File: README.md (hunk 3 above)
Comment (SlothOfAnarchy) [reply README.md]: I'm sure he just forgot the pacman tasks ;)

File: tasks/main.yml [hunk 5]
```diff
@@ -1,2 +1,17 @@
 ---
-# tasks file for .
\ No newline at end of file
+- name: Update Cache
+  apt:
+    update_chache: yes
+  changed_when: false
+
+- name: Install redis-server
+  apt:
```
Comment (SlothOfAnarchy) [inline tasks/main.yml]: These tasks fail on Arch Linux. Maybe add a pkg manager check?

File: README.md (hunk 2 above)
Comment (SlothOfAnarchy) [reply README.md]: Does every single var have a default value? Is nothing mandatory?
I guess the user has to go through the template file and read the comments there to understand and configure...

File: templates/redis.conf.j2 [hunk 6]
```diff
@@ -0,0 +1,1112 @@
+# Redis configuration file example.
+#
```
Comment (SlothOfAnarchy) [inline templates/redis.conf.j2]: At least add `ansible_managed` to this monster.

File: templates/redis.conf.j2 [hunk 7]
```diff
@@ -0,0 +1,1112 @@
...
+# without asking (in a push way), but just after a request, so only
+# asynchronous clients may create a scenario where data is requested faster
+# than it can read.
+#
+# Instead there is a default limit for pubsub and slave clients, since
+# subscribers and slaves receive data in a push fashion.
+#
+# Both the hard or the soft limit can be disabled by setting them to zero.
+client-output-buffer-limit {{ redis_client_output_buffer_limit }}
+client-output-buffer-limit {{ redis_client_output_buffer_limit }}
```
Comment (SlothOfAnarchy) [inline templates/redis.conf.j2]: I hope this is not intended.

**Comment by SlothOfAnarchy** [comment]
Also, meta information as well as a repo description and tags are missing

File: tasks/main.yml [hunk 8]
```diff
@@ -1,2 +1,17 @@
...
+    name: redis-server
+
+- name: Copy Redis conf
```
Comment (SlothOfAnarchy) [inline tasks/main.yml]: Is there a service that should be restarted if this file changes?

File: README.md (hunk 2 above)
Comment (Scriptkiddi) [reply README.md]: yes everthing is default you can download and run redis without configuring anything

File: templates/redis.conf.j2 (hunk 7 above)
Comment (Scriptkiddi) [reply templates/redis.conf.j2]: it is not

File: tasks/main.yml (hunk 8 above)
Comment (Scriptkiddi) [reply tasks/main.yml]: you are right forgot about that

Comment (Scriptkiddi) [reply tasks/main.yml]: done

File: templates/redis.conf.j2 (hunk 6 above)
Comment (Scriptkiddi) [reply templates/redis.conf.j2]: done

File: tasks/main.yml (hunk 5 above)
Comment (Scriptkiddi) [reply tasks/main.yml]: done

File: README.md (hunk 4 above)
Comment (Scriptkiddi) [reply README.md]: done

File: README.md (hunk 3 above)
Comment (Scriptkiddi) [reply README.md]: added pacman tasks

File: defaults/main.yml (hunk 1 above)
Comment (Scriptkiddi) [reply defaults/main.yml]: default right out of the redis conf

**Comment by Scriptkiddi** [comment]
pls rereview

File: tasks/main.yml [hunk 9]
```diff
@@ -3,10 +3,17 @@
...
     update_chache: yes
   changed_when: false
+  when: ansible_pkg_mgr == "apt"
 
 - name: Install redis-server
   apt:
     name: redis-server
+  when: ansible_pkg_mgr == "apt"
+
+- name: Install required packages (pacman)
```
Comment (SlothOfAnarchy) [inline tasks/main.yml]: update pacman cache

File: tasks/main.yml [hunk 10]
```diff
@@ -15,3 +22,19 @@
...
+- name: Copy Redis conf 
+  template:
+    src: redis.conf.j2
+    dest: /etc/redis.conf
+    owner: root
+    group: root
+    mode: 0600
+  when: ansible_pkg_mgr == "pacman"
+
+- name: Restart redis service
```
Comment (SlothOfAnarchy) [inline tasks/main.yml]: use a handler for this

Comment (Scriptkiddi) [reply tasks/main.yml]: no there is no advantage to add a handler since its only called one time and also enables the service

Comment (SlothOfAnarchy) [reply tasks/main.yml]: With the current implementation, the service will be restarted each time the playbook runs, even if there is no need to.
The handler would run every time the config changes, including the first time this role is executed.

**Comment by Scriptkiddi** [comment]
added handler @SlothOfAnarchy psl rereview

**Comment by SlothOfAnarchy** [comment]
Just add the repo description and tags then I think it's good to go, pending review from @dasJ.

**Comment by Scriptkiddi** [comment]
"Just add the repo description and tags " @SlothOfAnarchy What do you mean by that already edited meta.yml and added all of it

**Comment by SlothOfAnarchy** [comment]
I mean the repository description on github. I've sent you a picture.

**Comment by Scriptkiddi** [comment]
@SlothOfAnarchy @dasJ kann das bitte jemand mal reviewn

**Comment by SlothOfAnarchy** [comment]
Why is this so urgent?
I have no objections to this role, but I'm not sure @dasJ s issues are resolved satisfactory to him.

File: defaults/main.yml (hunk 1 above)
Comment (dasJ) [reply defaults/main.yml]: But will this also work when I update redis?

Comment (Scriptkiddi) [reply defaults/main.yml]: why should it not work its a socket path redis does not read shit from it

Comment (Scriptkiddi) [reply defaults/main.yml]: I can change it to /var/run/redis.pid

Comment (dasJ) [reply defaults/main.yml]: Please don't make me responsible when our redis breaks

**Comment by dasJ** [comment]
Btw, if you just look it up on the internet, you will find that 6379 is Redis' default port.

**Pull request merged**